pkg/webhook/pod/pods.go

```diff
@@ -186,12 +201,20 @@ func (pc *PodAdmissionControl) admitDeletePods(name, namespace string) *admissio
 		return util.ARSuccess()
 	}
 
+	payload := &admitPayload{
+		pod:              pod,
+		tc:               tc,
+		ownerStatefulSet: ownerStatefulSet,
+	}
+
 	if l.IsPD() {
 		pdClient := pc.pdControl.GetPDClient(pdapi.Namespace(namespace), tcName, tc.Spec.EnableTLSCluster)
-		return pc.admitDeletePdPods(pod, ownerStatefulSet, tc, pdClient)
+		payload.pdClient = pdClient
+		return pc.admitDeletePdPods(payload)
 	} else if l.IsTiKV() {
 		pdClient := pc.pdControl.GetPDClient(pdapi.Namespace(namespace), tcName, tc.Spec.EnableTLSCluster)
-		return pc.admitDeleteTiKVPods(pod, ownerStatefulSet, tc, pdClient)
+		payload.pdClient = pdClient
```

Comment:
the `payload.pdClient` is same with the PD section. It is better to add it to `payload` definition. 

	payload := &admitPayload{
		pod:              pod,
		tc:               tc,
		ownerStatefulSet: ownerStatefulSet,
		pdClient: pc.pdControl.GetPDClient(pdapi.Namespace(namespace), tcName, tc.Spec.EnableTLSCluster),
	}

Reply:
updated.